source/cursorManager.py

```diff
@@ -126,7 +126,14 @@ def _set_selection(self, info):
 		vision.handler.handleCaretMove(self)
 
 	def _caretMovementScriptHelper(self,gesture,unit,direction=None,posConstant=textInfos.POSITION_SELECTION,posUnit=None,posUnitEnd=False,extraDetail=False,handleSymbols=False):
-		oldInfo=self.makeTextInfo(posConstant)
+		oldInfo = self.makeTextInfo(posConstant)
+		# #5549 - Make sure oldInfo is not collapsed when going to top/bottom
+		# of page so that the deselection is announced.
+		oldSelection = self.makeTextInfo(textInfos.POSITION_SELECTION)
+		isJumpToTopOrBottom = posConstant == textInfos.POSITION_FIRST or posConstant == textInfos.POSITION_LAST
+		if not oldSelection.isCollapsed and isJumpToTopOrBottom:
+			oldInfo.isCollapsed = False
+
 		info=oldInfo.copy()
 		info.collapse(end=self.isTextSelectionAnchoredAtStart)
 		if self.isTextSelectionAnchoredAtStart and not oldInfo.isCollapsed:

```

Comment:
Is this going to be affected and should it be? What happens if you jump to the start/end of the page while the text selection is anchored at the start? Could this be included in the system test?

If this shouldn't be affected, maybe we should have `hasSelectionChanged = oldInfo.isCollapsed or (not oldSelection.isCollapsed and isJumpToTopOrBottom)` and use that boolean to check whether to `speakSelectionChange`